Production_scripts/Costs-02-district_nursing.R

```diff
@@ -0,0 +1,368 @@
+# Check the numbers in this file as some data completeness issues may mean the numbers can't be used to create costs.
+
+
+
+## data ##
+dn_raw_costs <- readxl::read_xlsx(
+  paste0(get_slf_dir(), "/Costs/DN_Costs.xlsx")
+)
+
+
+# change 1718 type to numeric - reads in as a character
+dn_raw_costs <-
+  dn_raw_costs %>%
+    mutate(`1718_Cost` = as.numeric(`1718_Cost`))
+
+
+dn_raw_costs <-
+  dn_raw_costs %>%
+    pivot_longer(
+      ends_with("Cost"),
+      names_to = "year_cost",
+      values_to = "cost"
+  )
+
+
+
+dn_raw_costs <-
+  dn_raw_costs %>%
+  # create year variable
+  mutate(year = substr(year_cost, 1, 4)) %>%
+  # remove variable year_cost
+  select(-year_cost) %>%
+  # sort by year and HB2019
+  arrange(year, HB2019)
+
+
+
+## save outfile ##
+# .zsav
+#haven::write_sav(dn_raw_costs,
+#                 paste0(get_slf_dir(), "/Costs/raw_dn_costs.sav"),
+#                 compress = TRUE
+#)
+
+
+
+# ----------------------------------------------------------
+
+## Get District Nursing file extracted from BOXI  ##
+
+dn_raw_costs_contacts <- readr::read_csv(paste0(get_slf_dir(), "/Costs/DN-Contacts-Numbers-for-Costs.csv"))
+
+# create year variable as fy
+dn_raw_costs_contacts <-
+  dn_raw_costs_contacts %>%
+  mutate(year = convert_year_to_fyyear(`Contact Financial Year`)) %>%
+  # rename TreatmentNHSBoardCode
+  rename(HB2019 = "Treatment NHS Board Code 9",
+         number_of_contacts = "Number of Contacts") %>%
+  # sort by year and HB2019
+  arrange(year, HB2019)
+
+
+
+## match files ##
+
+# run if not currently loaded from above #
+# read raw costs
+# dn_raw_costs <- haven::read_sav(paste0(get_slf_dir(), "/Costs/raw_dn_costs.sav"))
+
+
+# match
+dn_raw_costs_contacts <-
+  dn_raw_costs_contacts %>%
+  full_join(dn_raw_costs, by = c("HB2019", "year"))
+
+
+
+## save outfile ##
+# .zsav
+#haven::write_sav(dn_raw_costs_contacts,
+#                 paste0(get_slf_dir(), "/Costs/raw_dn_costs_with_contacts.sav"),
+#                 compress = TRUE
+#)
+
+
+
+
+# ----------------------------------------------------------
+
+
+## Calculate population cost for NHS Highland with HSCP population ratio. ##
+# Of the two HSCPs, Argyll and Bute provides the District Nursing data which is 27% of the population.
+
+lookup <- haven::read_sav(paste0(get_lookups_dir(), "/Populations/Estimates/HSCP2019_pop_est_1981_2020.sav"))
+
+
+# Select only the HSCPs for NHS Highland & years since 2015
+lookup <-
+  lookup %>%
+  filter(HSCP2019 == "S37000004" | HSCP2019 == "S37000016") %>%
+  filter(Year > 2015 | Year == 2015)
+
+
+# Create year as FY = YYYY from CCYY.
+lookup <-
+  lookup %>%
+  rename(calendar_year = Year) %>%
+  mutate(year = convert_year_to_fyyear(calendar_year))
+
+
+## outfile ##
+outfile <-
+  lookup %>%
+  select(year, HSCP2019, Pop) %>%
+  group_by(year, HSCP2019) %>%
+  summarise(pop = sum(Pop))
+
+
+
+# HSCP name
+outfile <-
+  outfile %>%
+  mutate(HSCPName = hscp_to_hscpnames(HSCP2019))
```

Comment:
Code is breaking here. Where is the function hscp_to_hscpnames? Can you find this and test it works. Then test the rest of the script is working please? :)

Reply:
function ` hscp_to_hscpnames` is on this branch in `convert_codes_to_names.R` should load when load_all() is done on this branch